src/shogun/clustering/KMeansLloydImpl.cpp

```diff
@@ -37,107 +49,117 @@ void CKMeansLloydImpl::Lloyd_KMeans(int32_t k, CDistance* distance, int32_t max_
 	bool vfree=false;
 	float64_t* vec=NULL;
 
+	/* Iterations */
 	while (changed && (iter<max_iter))
 	{
 		iter++;
 		if (iter==max_iter-1)
 			SG_SWARNING("kmeans clustering changed throughout %d iterations stopping...\n", max_iter-1)
 
-		if (iter%1000 == 0)
-			SG_SINFO("Iteration[%d/%d]: Assignment of %i patterns changed.\n", iter, max_iter, changed)
 		changed=0;
+		rhs_mus->copy_feature_matrix(mus);
+
+		if (distance->get_distance_type() == D_EUCLIDEAN)
+			((CEuclideanDistance*) distance)->precompute_rhs_squared_norms();
+		
+		/* Assigment step : Assign each point to nearest cluster */
+		for (int32_t i=0; i<lhs_size; i++)
+		{ 
+			const int32_t cl_list_i=cl_list[i];
+			int32_t min_cl, j;
+			float64_t min_dist, dist;
+			
+			min_cl=0;
+		   	min_dist=distance->distance(i,0);
+			for (j=1; j<k; j++)
+			{
+				dist=distance->distance(i,j);
```

Comment:
yes for that I have to store distances in a vector though. and `min` does similar loops inside.